### DISCUSSION TITLE
Load font(s) dynamically based on data with next/font

### DISCUSSION THREAD

**Discussion by JonasGruenwald** [vercel/next.js#50687] [Help]
### Summary

I have a backend that contains a dynamic list of fonts with associated font files, I want to load these with next/fonts at build time.
I have no interest in SSR, I would only build my site as a static export, and of course rebuild every time there is a change in my backend.
Ideally in my layout component, I would like to make an async request and then based on the result, call `localFont` from `next/font/local` in a loop for each font, but I think the only way this would be possible is with a top level await call, which from what I understand is not really supported in next.js.
Does anyone have any ideas on how to accomplish this?

**Comment by sdoomz** [comment]
One option I could think of is creating a local-font.json config during build or start of your application. You can go to server and fetch fonts and create something like this:

**local-font.json**
```json
{
  "src": [
    {
      "path": "./Roboto-Regular.woff2",
      "weight": "400",
      "style": "normal"
    },
    {
      "path": "./Roboto-Italic.woff2",
      "weight": "400",
      "style": "italic"
    },
  ]
}
```

**_app.ts**
```ts
import localFont from 'next/font/local';
import localFontConfig from "../local-font.json"
 
// Font files can be colocated inside of `pages`
const myFont = localFont(localFontConfig);
 
export default function MyApp({ Component, pageProps }) {
  return (
    <main className={myFont.className}>
      <Component {...pageProps} />
    </main>
  );
}
```

To achieve this you can create prebuild script, like this:

**package.json:**
```
...
"build": "node ./create-local-font-config.js && next build"
...
```

**Reply by JonasGruenwald** [reply]
I actually tried this approach now, and unfortunately it does not work, since apparently localFont has to be assigned to a const in module scope, but I need to call it from a loop, since I have an array of fonts to load.
I suppose I will have to load the fonts on my own, without next/font

**Comment by JonasGruenwald** [comment]
Yeah I also considered this, I thought perhaps next has some way to integrate this without an extra build step but I suppose I can just do that too yeah
